client/merge_client.go

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright 2023 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package client
+
+import (
+	"context"
+	"errors"
+	"net"
+	"runtime/debug"
+
+	"github.com/bytedance/gopkg/cloud/metainfo"
+
+	"github.com/cloudwego/kitex/client/callopt"
+	"github.com/cloudwego/kitex/internal/client"
+	internal_server "github.com/cloudwego/kitex/internal/server"
+	"github.com/cloudwego/kitex/internal/stats"
+	"github.com/cloudwego/kitex/pkg/consts"
+	"github.com/cloudwego/kitex/pkg/diagnosis"
+	"github.com/cloudwego/kitex/pkg/endpoint"
+	"github.com/cloudwego/kitex/pkg/kerrors"
+	"github.com/cloudwego/kitex/pkg/klog"
+	"github.com/cloudwego/kitex/pkg/proxy"
+	"github.com/cloudwego/kitex/pkg/remote/bound"
+	"github.com/cloudwego/kitex/pkg/rpcinfo"
+	"github.com/cloudwego/kitex/pkg/rpcinfo/remoteinfo"
+	"github.com/cloudwego/kitex/pkg/rpctimeout"
+	"github.com/cloudwego/kitex/pkg/serviceinfo"
+	"github.com/cloudwego/kitex/pkg/utils"
+)
+
+var localAddr net.Addr
+
+func init() {
+	localAddr = utils.NewNetAddr("tcp", "127.0.0.1")
+}
+
+type MergeMetaHandler func(ctx, svrCtx context.Context) context.Context
+
+type mergeClient struct {
+	svcInfo *serviceinfo.ServiceInfo
+	mws     []endpoint.Middleware
+	eps     endpoint.Endpoint
+	opt     *client.Options
+
+	inited bool
+	closed bool
+
+	// server info
+	serverEps endpoint.Endpoint
+	serverOpt *internal_server.Options
+
+	mergeMetaHandler MergeMetaHandler
+}
+
+type ServerMethod interface {
+	Endpoints() endpoint.Endpoint
+	Option() *internal_server.Options
+	GetServiceInfo() *serviceinfo.ServiceInfo
+}
+
+// NewMergeClient creates a kitex.Client with the given ServiceInfo, it is from generated code.
+func NewMergeClient(svcInfo *serviceinfo.ServiceInfo, s ServerMethod, opts ...Option) (Client, error) {
+	if svcInfo == nil {
+		return nil, errors.New("NewClient: no service info")
```

Comment:
fix NewClient

Reply:
It will conflict with common client because they are in the same pkg.